### PR TITLE
Story/145 tags filter removes too many items

### PR DESCRIPTION
**Pull request by redcatbear**

### CODE REVIEW

File: src/test/java/org/itsallcode/openfasttrace/cli/ITestCliWithFilter.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        this.specFile = this.tempFolder.newFile("spec.md");
+        writeTextFile(this.specFile, SPECIFICATION);
+        this.exit.expectSystemExitWithStatus(0);
+    }
+
+    // [itest->dsn~filtering-by-tags-during-import~1]
+    @Test
+    public void testWithoutFilter()
+    {
+        this.exit.checkAssertionAfterwards(new Assertion()
```
Comment (kaklakariada) [inline src/test/java/org/itsallcode/openfasttrace/cli/ITestCliWithFilter.java]: Is it possible to use a lambda expression here? Then `ITestCliWithFilter.` should not be necessary.
If lambdas are not possible as a parameter: extract a method that accepts a lambda.

File: src/test/java/org/itsallcode/openfasttrace/cli/TestCliArguments.java [hunk 2]
```diff
@@ -182,12 +182,23 @@ public void testSetI()
...
                 containsInAnyOrder("client", "server"));
     }
-}
+
+    // [utest->dsn~filtering-by-tags-or-no-tags-during-import~1]
+    @Test
+    public void testSetWantedTagsIncludingNone()
+    {
+        final String value = "_,client,server";
+        assertThat(BEFORE_SETTER, this.arguments.getWantedTags(), emptyIterable());
```
Comment (kaklakariada) [inline src/test/java/org/itsallcode/openfasttrace/cli/TestCliArguments.java]: You could move this to the setup method.

File: src/test/java/org/itsallcode/openfasttrace/testutil/AbstractSpecificationFileBasedTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+public class AbstractSpecificationFileBasedTest
+{
+
+    protected void writeTextFile(final File file, final String content) throws IOException
+    {
+        final PrintWriter writer = new PrintWriter(file);
```
Comment (kaklakariada) [inline src/test/java/org/itsallcode/openfasttrace/testutil/AbstractSpecificationFileBasedTest.java]: Is it possible to specify the encoding for the print writer? Else the encoding might depend on the platform.

File: src/test/java/org/itsallcode/openfasttrace/cli/ITestCliWithFilter.java (hunk 1 above)
Comment (redcatbear) [reply src/test/java/org/itsallcode/openfasttrace/cli/ITestCliWithFilter.java]: Fixed: with pleasure. :-)

File: src/test/java/org/itsallcode/openfasttrace/cli/TestCliArguments.java (hunk 2 above)
Comment (redcatbear) [reply src/test/java/org/itsallcode/openfasttrace/cli/TestCliArguments.java]: Improved by moving to separate test.

File: src/test/java/org/itsallcode/openfasttrace/testutil/AbstractSpecificationFileBasedTest.java (hunk 3 above)
Comment (redcatbear) [reply src/test/java/org/itsallcode/openfasttrace/testutil/AbstractSpecificationFileBasedTest.java]: Done.

**Pull request merged**